packages/docusaurus/src/webpack/base.ts

```diff
@@ -63,20 +61,40 @@ export function createBaseConfig(
   isServer: boolean,
   minify: boolean = true,
 ): Configuration {
-  const {outDir, siteDir, baseUrl, generatedFilesDir, routesPaths} = props;
-
+  const {
+    outDir,
+    siteDir,
+    baseUrl,
+    generatedFilesDir,
+    routesPaths,
+    siteMetadata,
+  } = props;
   const totalPages = routesPaths.length;
   const isProd = process.env.NODE_ENV === 'production';
   const minimizeEnabled = minify && isProd && !isServer;
   const useSimpleCssMinifier = process.env.USE_SIMPLE_CSS_MINIFIER === 'true';
 
   const fileLoaderUtils = getFileLoaderUtils();
 
+  const name = isServer ? 'server' : 'client';
+  const mode = isProd ? 'production' : 'development';
+
   return {
-    mode: isProd ? 'production' : 'development',
+    mode,
+    name,
+    cache: {
+      type: 'filesystem',
+      name: `${name}-${mode}-${props.i18n.currentLocale}`,
```

Comment:
@alexander-akait I've opened a dedicated issue here to see if it's possible to leverage the cache in a better way for our i18n system. I thought it would be simpler to expose the use-case in a dedicated issue: https://github.com/webpack/webpack/issues/13034